api/db/database-builder/database-builder.js

```diff
@@ -128,14 +128,28 @@ class DatabaseBuilder {
       'knex_migrations_lock',
       'view-active-organization-learners',
     );
-    const tables = _.map(sortedTables, (tableToDelete) => `"${tableToDelete}"`).join();
-    // eslint-disable-next-line knex/avoid-injections
-    return this.knex.raw(`TRUNCATE ${tables}`);
+
+    const publicTables = sortedTables
+      .filter((tableName) => !tableName.includes('pgboss'))
+      .map((tableName) => `"${tableName}"`)
+      .join();
+    const pgbossTables = sortedTables
+      .filter((tableName) => tableName.includes('pgboss'))
+      .map((tableName) => {
+        return tableName
+          .split('.')
+          .map((element) => `"${element}"`)
+          .join('.');
+      })
+      .join();
+
+    return Promise.all([this.knex.raw(`TRUNCATE ${publicTables}`), this.knex.raw(`TRUNCATE ${pgbossTables}`)]);
   }
 
   async _initTablesOrderedByDependencyWithDirtinessMap() {
     // See this link : https://stackoverflow.com/questions/51279588/sort-tables-in-order-of-dependency-postgres
-    const results = await this.knex.raw(`with recursive fk_tree as (
+    function _rawQuery(namespace) {
```

Comment:
Suggestion de renommage pour cette fonction : 
```suggestion
    function _createRawQueryForNamespace(namespace) {
```

Reply:
Et idéalement il serait bien de sortir cette fonction de la fonction dans laquelle elle est déclarée.

Reply:
Comme cette fonction est uniquement utilisée par `_initTablesOrderedByDependencyWithDirtinessMap` parce que la query qui va être utilisé n'est utile que pour cette fonction `_initTablesOrderedByDependencyWithDirtinessMap`. 

Si tu souhaites la sortir, il faut alors revoir complètement le nom de la fonction `_createInitTablesOrderedByDependencyRawQueryForNamespace`

Reply:
À la réflexion je suis d'accord pour garder la fonction à l'intérieur avec le nommage et la signature suivante : `function _constructRawQuery(namespace)`

À ma connaissance les fonctions déclarées dans des fonctions sont difficilement/pas optimisables par le JIT.

Reply:
Pour plus d'information, la fonction `_initTablesOrderedByDependencyWithDirtinessMap` ne sera appelée qu'une seule fois lors de l'exécution du 1er commit (`databaseBuilder.commit()`), pour toute l'exécution des tests d'acceptance ou d'intégration.